File: solutions/go/diamond/1/diamond.go
package diamond

import (
    "bytes"
    "errors"
)

func Gen(char byte) (string, error) {
    if char > 'Z' || char < 'A' {
        return "", errors.New("Bad symbol!")
    }
	n := (int(char) - int('A')) * 2 + 1
    data := make([][]byte, n)
    for i := range data {
        data[i] = make([]byte, n)
        for j := range data[i] {
            data[i][j] = ' '
        }
    }
    left, right := n / 2, n / 2
    
    for i := 0; i <= int(char-'A'); i++ {
        data[i][left] = byte('A' + i)
        data[i][right] = byte('A' + i)
        left--
        right++
    }
    
    left = 1
    right = n - 2
    for i := n / 2 + 1; i < n; i++ {
        symIndex := n - 1 - i
        data[i][left] = data[symIndex][left]
        data[i][right] = data[symIndex][right]
        left++
        right--
    }

    result := string(bytes.Join(data, []byte{'\n'}))
    return result, nil
}

File: solutions/python/diamond/1/diamond.py
def rows(letter):
    if not letter.isalpha():
        raise ValueError("Bad symbol!")
    num = (ord(letter) - ord('A')) * 2 + 1
    data = [bytearray(b' ' * num) for _ in range(num)]
    left, right = num // 2, num // 2

    for index in range(0, ord(letter) - ord('A') + 1):
        current_char_code = ord('A') + index
        data[index][left] = current_char_code
        data[index][right] = current_char_code
        left-=1
        right+=1

    for index in range(num // 2 + 1, num):
        sym_index = num - 1 - index
        data[index] = data[sym_index].copy()

    return [row.decode('utf-8') for row in data]
